gemac/management.py

```diff
@@ -0,0 +1,155 @@
+    mdio_interface.mdiodriver = mdio_interface.mdioio.driver()
+
+    configregisters = [Signal(intbv(0)[32:0]) for _ in range(10)]
+
+    def getregisternumber(addr):
+        if addr >= 0x200 and addr <= 0x23F:
+            return rx0
+        elif addr >= 0x240 and addr <= 0x27F:
+            return rx1
+        elif addr >= 0x280 and addr <= 0x2BF:
+            return tx
+        elif addr >= 0x2C0 and addr <= 0x2FF:
+            return flow
+        elif addr >= 0x340 and addr <= 0x37F:
+            return managementreg
+        elif addr >= 0x380 and addr <= 0x383:
+            return ucast0
+        elif addr >= 0x384 and addr <= 0x387:
+            return ucast1
+        elif addr >= 0x388 and addr <= 0x38B:
+            return addrtable0
+        elif addr >= 0x38C and addr <= 0x38F:
+            return addrtable1
+        elif addr >= 0x390 and addr <= 0x393:
+            return addrfiltermode
+        else:
+            raise Exception
+
+    @always_seq(host_interface.clk.posedge, reset=None)
+    def readConfig():
+        if((not host_interface.miimsel) and host_interface.regaddress[9] and
+                host_interface.opcode[1]):
+            host_interface.rddata.next = \
+                configregisters[getregisternumber(host_interface.regaddress)]
+
+    @always_seq(host_interface.clk.posedge, reset=None)
+    def writeConfig():
+        if((not host_interface.miimsel) and host_interface.regaddress[9] and
+                (not host_interface.opcode[1])):
+            configregisters[getregisternumber(host_interface.regaddress)]\
+                .next = host_interface.wrdata
+
+    # @TODO: Address Table Read.
+
+    @always_comb
+    def mdiodriver():
+        mdio_interface.mdiodriver.next = mdioout if mdiotri else None
+        mdioindriver.next = mdio_interface.mdioio if not mdiotri else None
+        # mdioin.next = False if mdio_interface.mdioio is None else mdio_interface.mdioio
+
+    @always_seq(host_interface.clk.posedge, reset=None)
+    def mdcdriver():
+        global hostclk_count
+        clkDiv = (configregisters[managementreg][5:] + 1)  # * 2
+        hostclk_count += 1
+        if hostclk_count == clkDiv:
+            mdio_interface.mdc.next = not mdio_interface.mdc
+            hostclk_count = 0
+
+    # May req to recheck trigerring signal according to Page 87, Xilinx UG144
+    @always_seq(host_interface.clk.posedge, reset=None)
+    def mdioinitiate():
+        if host_interface.hostreq and host_interface.miimsel:
+            host_interface.miimrdy.next = 0
+            mdiodata = mdioData(host_interface.opcode,
+                                host_interface.regaddress,
+                                host_interface.wrdata)
+            # yield host_interface.clk.posedge
+            if host_interface.opcode[1]:  # Read Operation
+                host_interface.rddata.next = mdioread(mdiodata)
+            elif not host_interface.opcode[1]:  # Write Operation
+                mdiowrite(mdiodata)
```

Comment:
@tompdillon made me aware about it. And am making changes to making it convertible. Added a test to check the convertibility too.